samples/nrf9160/gps/Kconfig

```diff
@@ -17,6 +17,7 @@ choice GPS_SAMPLE_ANTENNA
 
 config GPS_SAMPLE_ANTENNA_ONBOARD
 	bool "Use onboard antenna"
+	depends on !BOARD_CIRCUITDOJO_FEATHER_NRF9160NS
```

Comment:
@mniestroj We had an internal discussion regarding this and we've decided that we should not reference 3rd party boards in the NCS codebase - this simply doesn't scale for us. We don't have the means to verify that samples are operational on 3rd party devices on each release, and having them mentioned in the Kconfig files would suggest so.

Could you drop the commit adding the `BOARD_CIRCUITDOJO_FEATHER_NRF9160NS` support then? With other changes in this PR, it should be pretty straightforward to configure the sample for custom boards with only configuration changes.

Reply:
>We had an internal discussion regarding this and we've decided that we should not reference 3rd party boards in the NCS codebase - this simply doesn't scale for us. We don't have the means to verify that samples are operational on 3rd party devices on each release, and having them mentioned in the Kconfig files would suggest so.

I understand that you do not have means for maintaining 3rd party boards. The alternative is to copy whole sample and adjust it for 3rd party boards. This is exactly what happens in https://github.com/circuitdojo/nrf9160-feather-examples-and-drivers/tree/v1.5.x/samples/gps. The result is that sdk-nrf's sample is compatible with nRF9160 DK and nrF9160 Thingy, while Circuitdojo's copy is an outdated GPS example which additionally supports nRF9160 Feather. Unfortunately this approach doesn't scale in another way, resulting in one repository per one board vendor. I think it is less work to maintain all boards in single repository, which is exactly what happens in mainline Zephyr. As nRF9160 GPS and LTE is supported only here, then in my opinion this is the repository in which nRF9160 boards should be supported. This gives users (and potential Nordic customers) one entry point for trying all boards. The documentation can simply state that Nordic is responsible for nRF9160 DK and Thingy, while other boards are supported by community.

>Could you drop the commit adding the BOARD_CIRCUITDOJO_FEATHER_NRF9160NS support then?

Created new #5079 PR, as the subject of intended changes is much different from original subject of this PR.

>With other changes in this PR, it should be pretty straightforward to configure the sample for custom boards with only configuration changes, that could be applied locally.

Pretty straighforward if we know what to set and to which values. Question is where should that information be documented and maintained? As written above, currently each vendor is responsible for creating new repository with copy-pasted samples and putting changes there. Alternatively board vendor can document what to configure before using samples from sdk-nrf, which complicates things for new users.